src/main/java/org/ow2/proactive/catalog/util/parser/WorkflowParser.java

```diff
@@ -142,38 +148,63 @@ private void addVariableIfNotNullAndModelIfNotEmpty(
         }
         if (checkIfNotNull(name, model) && checkIfNotEmpty(name, model)) {
             keyValueMapBuilder.add(new KeyValueLabelMetadataEntity(name, model, ATTRIBUTE_VARIABLE_MODEL_LABEL));
-            addDependencyOn(keyValueMapBuilder, value, model);
+            addDependsOn(keyValueMapBuilder, value, model);
         }
     }
 
-    private void addDependencyIfCatalogObjectModelExist(
-            ImmutableList.Builder<KeyValueLabelMetadataEntity> keyValueMapBuilder, TaskVariable taskVariable) {
+    private void addDependsOnIfCatalogObjectModelExistOnTaskVariable(
+            ImmutableSet.Builder<KeyValueLabelMetadataEntity> keyValueMapBuilder, TaskVariable taskVariable) {
         String name = taskVariable.getName();
         String value = taskVariable.getValue();
         String model = taskVariable.getModel();
         if (checkIfNotNull(name, model) && checkIfNotEmpty(name, model)) {
-            addDependencyOn(keyValueMapBuilder, value, model);
+            addDependsOn(keyValueMapBuilder, value, model);
         }
 
     }
 
-    private void addDependencyOn(ImmutableList.Builder<KeyValueLabelMetadataEntity> keyValueMapBuilder, String value,
+    private void addDependsOn(ImmutableSet.Builder<KeyValueLabelMetadataEntity> keyValueMapBuilder, String value,
             String model) {
         if (model.equalsIgnoreCase(CATALOG_OBJECT_MODEL)) {
-            keyValueMapBuilder.add(new KeyValueLabelMetadataEntity(DEPENDS_ON, value, ATTRIBUTE_DEPENDENCIES_LABEL));
+            if (model.equalsIgnoreCase(CATALOG_OBJECT_MODEL)) {
+                keyValueMapBuilder.add(new KeyValueLabelMetadataEntity(getNameAndBucketFromDependsOn(value),
+                        getRevisionFromDependsOn(value).orElse(LATEST_VERSION),
+                        ATTRIBUTE_DEPENDS_ON_LABEL));
+            }
+        }
+    }
+
+    private String getNameAndBucketFromDependsOn(String calledWorkflow) {
+        Pattern pattern = Pattern.compile(CATALOG_OBJECT_MODEL_REGEXP);
+        Matcher matcher = pattern.matcher(calledWorkflow);
+        if (!(calledWorkflow.matches(CATALOG_OBJECT_MODEL_REGEXP) && matcher.find())) {
+            throw new RuntimeException(String.format("Impossible to parse the PA:CATALOG_OBJECT: %s, parsing error when getting the bucket and the workflow name",
+                    calledWorkflow));
+        } else {
+            return (matcher.group(1));
+
+        }
+    }
+
+
+    private Optional<String> getRevisionFromDependsOn(String calledWorkflow) {
+        try {
+            return Optional.of(calledWorkflow.split("/")[2]);
```

Comment:
why don't you use your regexp pattern to extract getRevisionFromDependsOn 
with group 3 for example?